### PR TITLE
zip_index method for Composable

### PR DESCRIPTION
**Pull request by AlirezaSohofi**
# Description

This pr would resolve the issue that shard keys are not sortable. 

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. 
List any dependencies that are required for this change.

Fixes # issue

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring including code style reformatting 
- [ ] Other (please describe):

# Checklist:

- [ ] I have read the [contributing guideline doc](https://squirrel-core.readthedocs.io/en/latest/usage/code_of_conduct.html) (external contributors only)
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have kept the PR small so that it can be easily reviewed  
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] All dependency changes have been reflected in the pip requirement files.

### CODE REVIEW

File: squirrel/iterstream/base.py [hunk 1]
```diff
@@ -211,6 +211,15 @@ def loop(self, n: t.Optional[int] = None) -> Composable:
         """
         return _LoopIterable(self, n)
 
+    def zip_index(self, pad_length: int = None) -> Composable:
+        """Zip the item in the stream with its index and yield Tuple[index, item]
```
Comment (pzdkn) [inline squirrel/iterstream/base.py:215]: Setting pad_length to None is the most "dangerous" option actually if the user expects the shards to be sorted.

We should warn/add to the docstring what will happen when the pad_length < length of longest sequences of digits

Comment (AlirezaSohofi) [reply squirrel/iterstream/base.py:215]: This method is generic on its own right, not specifically meant to solve the key ordering, so the docstring does not need to explain that. I'd say a section in the doc or a tutorial notebook is a better place for that.

File: requirements.txt [hunk 2]
```diff
@@ -545,6 +545,9 @@ isodate==0.6.1 \
 itsdangerous==2.1.2 \
     --hash=sha256:2c2349112351b88699d8d4b6b075022c0808887cb7ad10069318a8b0bc88db44 \
     --hash=sha256:5dbbc68b317e5e42f327f9021763545dc3fc3bfe22e6deb96aaf1fc38874156a
+jeepney==0.8.0 \
```
Comment (pzdkn) [inline requirements.txt:548]: I guess this should follow a change in the `requirements.in`

**Review by pzdkn — APPROVED** [review]
LGTM!

File: requirements.txt (hunk 2 above)
Comment (AlirezaSohofi) [reply requirements.txt:548]: This is not our direct dependency

**Pull request merged**